src/tests/join/hash_join-tests.cu

```diff
@@ -0,0 +1,678 @@
+   * @Param sort Option to sort the result. This is required to compare the result against the reference solution
+   */
+  /* ----------------------------------------------------------------------------*/
+  std::vector<result_type> compute_gdf_result(bool print = false, bool sort = true)
+  {
+    const int num_columns = std::tuple_size<multi_column_t>::value;
+
+    gdf_column left_result;
+    gdf_column right_result;
+
+    gdf_error result_error{GDF_SUCCESS};
+
+    gdf_column ** left_gdf_columns = gdf_raw_left_columns.data();
+    gdf_column ** right_gdf_columns = gdf_raw_right_columns.data();
+    // Use single column join when there's only a single column
+    if(num_columns == 1){
+      switch(op)
+      {
+        case join_op::LEFT:
+          {
+            result_error = gdf_left_join(num_columns,
+                                         left_gdf_columns,
+                                         right_gdf_columns,
+                                         &left_result, &right_result,
+                                         &ctxt);
+            break;
+          }
+        case join_op::INNER:
+          {
+            result_error = gdf_inner_join(num_columns,
+                                         left_gdf_columns,
+                                         right_gdf_columns,
+                                         &left_result, &right_result,
+                                         &ctxt);
+            break;
+          }
+        case join_op::OUTER:
+          {
+            result_error = gdf_outer_join_generic(gdf_raw_left_columns[0],
+                                                  gdf_raw_right_columns[0],
+                                                  &left_result, &right_result);
+            break;
+          }
+        default:
+          std::cout << "Invalid join method" << std::endl;
+          EXPECT_TRUE(false);
+      }
+
+    }
+    // Otherwise use the multicolumn join
+    else
+    {
+      gdf_column ** left_gdf_columns = gdf_raw_left_columns.data();
+      gdf_column ** right_gdf_columns = gdf_raw_right_columns.data();
+      switch(op)
+      {
+        case join_op::LEFT:
+          {
+            result_error = gdf_left_join(num_columns,
+                                         left_gdf_columns,
+                                         right_gdf_columns,
+                                         &left_result, &right_result,
+                                         &ctxt);
+            break;
+          }
+        case join_op::INNER:
+          {
+            result_error =  gdf_inner_join(num_columns,
+                                           left_gdf_columns,
+                                           right_gdf_columns,
+                                           &left_result, &right_result,
+                                           &ctxt);
+            //std::cout << "Multi column *inner* joins not supported yet\n";
+            //EXPECT_TRUE(false);
```

Comment:
please remove dead code

Reply:
All right!

Reply:
@scopatz This unit tests is never  used because all new unit tests is in join-tests.cu so I remove it.